matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/model/message/MessageLocationContent.kt

```diff
@@ -64,4 +64,12 @@ data class MessageLocationContent(
 ) : MessageContent {
 
     fun getBestGeoUri() = locationInfo?.geoUri ?: geoUri
+
+    /**
+     * @return true if location asset is different than LocationAssetType.SELF
+     */
+    fun isGenericLocation(): Boolean {
+        // Should behave like m.self if locationAsset is null
+        return locationAsset != null && locationAsset.type != LocationAssetType.SELF
+    }
```

Comment:
I think it will be simpler to test the opposite, so something like:
```kotlin
    fun isSelfLocation(): Boolean {
        // Should behave like m.self if locationAsset is null
        return locationAsset?.type == null || locationAsset.type == LocationAssetType.SELF
    }
```

Reply:
Done.